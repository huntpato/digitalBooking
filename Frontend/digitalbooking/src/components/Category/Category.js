import React from "react";
import { ProductsByCategory } from "../ProductsByCategory/ProductsByCategory";
import { useLocation, useHistory } from 'react-router-dom';
import useFetch from "use-http";
import styles from "./Category.module.css";

export const Category = () => {
  const location = useLocation();
  const path= location.pathname.split("/");
  const id = path[2];
  const options = { headers:{
    "X-Requested-With": "XMLHttpRequest"
  }};
  const url = `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/category/${path[2]}`;
  const { loading, data = [] } = useFetch(url, options, []);
  const { category, title, cat, titleContainer } = styles;
  let history = useHistory();
  
  
  return (
    <div className={cat}>
        <section className= {title}>
          <div className={titleContainer}>
            <h4 className= {category}>{data.name}</h4>
            <i className="fas fa-arrow-left" onClick={() => history.goBack()}></i>
          </div>
        </section>
          {loading && <p>Cargando Recomendaciones...</p>}  
          <ProductsByCategory category={id} name={data.name}/>
    </div>
  );
};