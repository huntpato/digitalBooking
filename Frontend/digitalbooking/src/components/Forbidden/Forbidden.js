import React from 'react'
import { Link } from "react-router-dom";
import checked from '../../assets/forbidden.png'
import styles from  './Forbidden.module.css'

export const Forbidden = (props) => {
    const {confirmation, success, title, info, button} = styles;
    const {mes1, mes2, link, btn} = props;
    return (
        <div className= { confirmation }>
            <section className= { success }>
                <img src= { checked } alt='Ícono de reserva realizada.'/>
                <h3 className= { title }>{mes1}</h3>
                <h5 className= { info }>{mes2}</h5>
                <Link to= {link}><button className={ button }>{btn}</button></Link>
            </section>
        </div>
    )
}

Forbidden.defaultProps = {
    mes1: "Login necesario",
    mes2: "Para acceder a esta página se requiere iniciar sesión.",
    link: '/login',
    btn: "Iniciar sesión"
};