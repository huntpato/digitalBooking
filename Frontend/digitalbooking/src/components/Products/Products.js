import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { NotFound } from './NotFound/NotFound'
import useFetch from "use-http";
import { ReCard } from "../ReCard/ReCard";
import styles from "./Products.module.css";

export const Products = () => {
  const place = localStorage.getItem("place");
  const date = localStorage.getItem("date");

  let startDate = date!== "," && new Date(date.split(",")[0]).toISOString().split("T")[0];
  let endDate =  date!== "," && new Date(date.split(",")[1]).toISOString().split("T")[0];
  let city= '';
 

  const setPlaceFormat = () => {
    if (place !== "false" && place !== null) {
      let splitPlace = place.replace(/,[^,]+$/, "");
      city = splitPlace.replace(/ /g, "%20");
    } 
  };
  setPlaceFormat();


  let url2= `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/reservations/search/${city}?startdate=${startDate}&enddate=${endDate}`;

 
  const options = {
    headers: {
      "X-Requested-With": "XMLHttpRequest",
    },
  };


  let url = date!== ","
    ? url2
    : "https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products";
  let searchDate = "";

  const setPlace = () => {
    if (url !== url2 && place !== "false" && place !== null) {
      url = `${url}/search?city=${city}`;
  }};
  setPlace();


  const setDate = () => {
    if (date) {
      let dates = date.split(/(\s+)/);
      searchDate = `${dates[4]} de ${dates[2]} del ${dates[6]} hasta el ${dates[22]} de ${dates[20]} del ${dates[24]}  `;
    }
  };
  setDate();

  const { loading, data = [] } = useFetch(url, options, []);
  console.log(data);
  const [products, setProducts] = useState(4);
  const {
    productsList,
    recommendation,
    recommendationCards,
    recommendationtitle,
    recommendationContainer,
    loadMore,
    title,
    placeInfo,
    titleContainer
  } = styles;
  let history = useHistory();

  const handleClick = () => {
    setProducts((prevProducts) => prevProducts + 4);
  };

 
  return (
    <div className={productsList}>
      <section className={title}>
        <div className={titleContainer}>
          <h4 className={placeInfo}>
            {" "}
            {place !== "false"
              ? `Resultados de tu búsqueda en ${place}`
              : `Sugerencias para tus próximas vacaciones`}
          </h4>{" "}
          <i className="fas fa-arrow-left" onClick={() => history.goBack()}></i>
        </div>
      </section>
      <div className={recommendationContainer}>
        <section className={recommendation}>
          <h4 className={recommendationtitle}>
            {date === ","
              ? `Productos disponibles`
              : `Productos disponibles para el día ${searchDate}`}
          </h4>
          <div className={recommendationCards}>
            {loading ? <p>Cargando Recomendaciones...</p> :
            data.length>=1 ? data.slice(0, products).map((product, index) => (
              <ReCard
                key={index}
                type={product.productCategory.name}
                title={product.productName}
                img={product.photoGallery[0]}
                alt={"Isla Fanta sia"}
                text={product.productDescription}
                loc={product.location}
                rate={product.review}
                id={product.id}
                services={product.services}
                latlong={product.coordinates}
              /> 
            )) : <NotFound />}
          </div>
          {data.lenght > 6 && (
            <button className={loadMore} onClick={handleClick}>
              Ver Mas
            </button>
          )}
        </section>
      </div>
      
    </div>
  );
};