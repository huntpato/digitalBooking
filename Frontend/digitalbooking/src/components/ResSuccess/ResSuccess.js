import React from 'react'
import { Link } from "react-router-dom";
import checked from '../../assets/checked.png'
import styles from  './ResSuccess.module.css'

export const ResSuccess = (props) => {
    const {confirmation, success, title, info, button} = styles;
    const { message, text, bMessage } = props;
    return (
        <div className= { confirmation }>
            <section className= { success }>
                <img src= { checked } alt='Ícono de reserva realizada.'/>
                <h3 className= { title }>{ message }</h3>
                <h5 className= { info }>{text}</h5>
                <Link to= '/'><button className={ button }>{ bMessage }</button></Link>
            </section>
        </div>
    )
}
