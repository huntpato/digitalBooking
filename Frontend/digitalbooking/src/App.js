import React, { useState, useMemo, useEffect } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Nav } from "./components/Nav/Nav";
import { Main } from "./components/Main/Main";
import { UserLogin } from "./components/UserLogin/UserLogin";
import { Category } from "./components/Category/Category";
import { Footer } from "./components/Footer/Footer";
import { SignUp } from "./components/SignUp/SignUp";
import { SingleProduct } from "./components/SingleProduct/SingleProduct";
import { ResSuccess } from "./components/ResSuccess/ResSuccess";
import { Reservation } from "./components/Reservation/Reservation";
import { Products } from "./components/Products/Products";
import { UserDesk } from "./components/UserDesk/UserDesk";
import { CreateProduct } from "./components/CreateProduct/CreateProduct";
import { NotFound } from "./components/NotFound/NotFound";
import { UserContext } from "./components/UserContext/UserContext";
import styles from "./App.module.css";
import { Forbidden } from "./components/Forbidden/Forbidden";

export const App = (props) => {
  const [user, setuser] = useState(null);
  const auth = user != null && JSON.parse(user).authorities[0];
  const [item, setItem] = useState(null); 
  const value = useMemo(() => ({ user, setuser }), [user, setuser]);
  const [showButton, setShowButton] = useState(false);
  const { toTop } = styles;
  const lastclear = localStorage.getItem('lastclear');
  const time_now  = (new Date()).getTime();
 
  useEffect(() => {
    window.addEventListener("scroll", () => {
      if (window.pageYOffset > 300) {
        setShowButton(true);
      } else {
        setShowButton(false);
      }
    });
  }, []);

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  useEffect(() => {

    if ((time_now - lastclear) > 1000 * 60 * 60 * 24) {

      localStorage.clear();
  
      localStorage.setItem('lastclear', time_now);
    }

    if(localStorage.getItem("data")){
      setuser(localStorage.getItem("data"))
    }

    if(user== null){
      setItem(localStorage.getItem("data"))
    } else{ 
      setItem(null);
    }  

  }, [item, lastclear, time_now]);

  


  return (
    <Router>
      <div className="App">
        <UserContext.Provider value={value}>
          <Nav />
          <Switch>
            <Route exact path="/" component={Main} />
            <Route exact path="/register" component={SignUp} />
            <Route exact path="/login" component={UserLogin} />
            <Route exact path="/login/:login_id" component={UserLogin} />
            <Route exact path="/logout" component={Main} />
            <Route exact path="/product" component={Products} />
            <Route exact path="/administration" render={ ()=> auth == "ADMIN" ?  <CreateProduct/> : <Forbidden mes1="Acceso no autorizado" mes2="No tiene las credenciales necesarias para poder acceder" link="/" btn="Ir al home"/>}/>
            <Route
              exact
              path="/product/:product_id"
              component={SingleProduct}
            />
            <Route exact path="/category/:categoy_id" component={Category} />
            <Route exact path="/reservation" component={ user ?Reservation : Forbidden } />
            {/*Ruta solo para pruebas, borrar antes de la entrega"*/}
            <Route
              exact
              path="/reservation/:reservation_id"
              component={ user ? Reservation : Forbidden }
            />
            <Route exact path="/confirmation" render={()=> <ResSuccess message= "¡Muchas gracias!" text="Su reserva fue realizada con éxito." bMessage= "OK" />} />
            <Route exact path="/verification" render={()=> <ResSuccess message= "¡Muchas gracias!" text="Su email ha sido verificado exitosamente." bMessage= "Volver al home"/>} />
            <Route exact path="/success" render={()=> <ResSuccess message= "¡Muchas gracias!" text="Tu propiedad se ha creado con éxito." bMessage= "Volver al home"/>} />
            <Route exact path="/preferences" component={ user ? UserDesk : Forbidden }/>
            <Route exact path="/cancellation" render={()=> <ResSuccess message= "Reserva cancelada" text="Tu reserva ha sido cancelada con éxito." bMessage= "Volver al home"/>} />
            <Route path="*" component={NotFound} />
          </Switch>
          {showButton && (
            <button onClick={scrollToTop} className={toTop}>
              <i className="fas fa-chevron-up"></i>
            </button>
          )}
          <Footer />
        </UserContext.Provider>
      </div>
    </Router>
  );
};
