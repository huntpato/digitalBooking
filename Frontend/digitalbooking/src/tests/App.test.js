import React from 'react';
import { MemoryRouter } from 'react-router';
import {shallow, mount} from 'enzyme';
import {App} from '../App'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';
import { ResSuccess } from '../components/ResSuccess/ResSuccess';
import { Category } from '../components/Category/Category';
import { Forbidden } from '../components/Forbidden/Forbidden';
import { SingleProduct } from '../components/SingleProduct/SingleProduct';
import { Products } from '../components/Products/Products';
import { UserLogin } from '../components/UserLogin/UserLogin';
import { SignUp } from '../components/SignUp/SignUp';

jest.mock('lottie-web')

describe('Pruebas en componente <App />', ()=> {

   
    test('Comprobar que <App /> renderiza correctamente.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), // Deprecated
              removeListener: jest.fn(), // Deprecated
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/']}><App /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
    })

    test('Comprobar que la ruta /register renderiza el componente Sign Up.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(), 
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={[ { pathname: '/register'}] }><SignUp /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h2').at(0).text()).toContain('Crear cuenta')
    })
    
    test('Comprobar que la ruta /login renderiza el componente UserLogin.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(), 
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={[ { pathname:'/login'}]}><UserLogin /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();

    })

    test('Comprobar que la ruta /logout renderiza el componente Main.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), // Deprecated
              removeListener: jest.fn(), // Deprecated
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/logout']}><App /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('main').hasClass('main')).toBe(true)
    })
    

    test('Comprobar que la ruta /administration renderiza el componente Forbidden.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/administration']}><Forbidden /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();

        
    })

    test('Comprobar que la ruta /product/1 renderiza el componente SingleProduct.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/product/1']}><SingleProduct /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('Comprobar que la ruta /category/1 renderiza el componente Category.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/category/1']}><Category /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        
    })

    
    test('Comprobar que la ruta /reservation/1 renderiza el componente Forbidden.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/reservation/1'] }><Forbidden /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('Comprobar que la ruta confirmation renderiza el componente ResSuccess.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/confirmation'] }><ResSuccess /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('Comprobar que la ruta verification renderiza el componente ResSuccess.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/verification'] }><ResSuccess message= "¡Muchas gracias!" text="Su email ha sido verificado exitosamente." bMessage= "Volver al home"/></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        
    })
  
     test('Comprobar que la ruta cancellation renderiza el componente ResSuccess.', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), 
              removeListener: jest.fn(),
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
        const wrapper= mount(<MemoryRouter initialEntries={['/cancellation'] }><ResSuccess /></MemoryRouter>)
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('Comprobar que aparezca la barra de búsqueda al hacer scroll', () => {
      const spyScrollTo = jest.fn();
      Object.defineProperty(global.window, 'scrollTo', { value: spyScrollTo });
      Object.defineProperty(global.window, 'scrollY', { value: 400 });

      const wrapper= mount(<MemoryRouter initialEntries={['/cancellation'] }><ResSuccess /></MemoryRouter>)
      expect( wrapper.find('input').at(0)).toMatchSnapshot();
    })
    

});