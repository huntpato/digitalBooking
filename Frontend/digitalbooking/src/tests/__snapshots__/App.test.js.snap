// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <App /> Comprobar que <App /> renderiza correctamente. 1`] = `
<div
  className="App"
>
  Array [
    <div
      className="menuhide"
    >
      <div>
        <section
          className="header"
        >
          <button
            className="logout"
            onClick={[Function]}
          >
            X
          </button>
          <h4>
            MENÚ
          </h4>
        </section>
        <section
          className="buttons"
        >
          <a
            className="button"
            href="/register"
            onClick={[Function]}
          >
            <h5>
              Crear cuenta
            </h5>
          </a>
          <a
            className="button"
            href="/login"
            onClick={[Function]}
          >
            <h5>
              Iniciar sesión
            </h5>
          </a>
        </section>
      </div>
      <div
        className="social"
      >
        <section
          className="header"
        />
        <div
          className="socialbar"
        >
          <i
            className="fab fa-facebook"
          />
          <i
            className="fab fa-linkedin-in"
          />
          <i
            className="fab fa-twitter"
          />
          <i
            className="fab fa-instagram"
          />
        </div>
      </div>
    </div>,
    <nav
      className="color1"
    >
      <img
        alt="Logo"
        onClick={[Function]}
        src="logo.png"
      />
      <section
        className="searchInput"
      >
        <i
          className="fas fa-search"
        />
        <input
          className="searchNav"
          placeholder="Encontrá tu lugar"
          type="text"
        />
      </section>
      <div
        className="registerButtons"
      >
        <a
          href="/register"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
           
          <button
            className="button"
          >
            Crear cuenta
          </button>
        </a>
        <a
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
           
          <button
            className="button"
          >
            Iniciar sesion
          </button>
        </a>
      </div>
      <img
        alt="ícono menú"
        className="hamburguesita"
        onClick={[Function]}
        src="hamburguer.svg"
      />
    </nav>,
  ]
  <main
    className="main"
  >
    <div
      className="search"
    >
      <p>
        Busca ofertas en hoteles, casas y mucho más
      </p>
      <section>
        <div
          className="container"
        >
          <input
            className="input"
            content="width=device-width, initial-scale=1, maximum-scale=1"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="¿A dónde vamos?"
            type="text"
            value=""
          />
          <div
            className="containerLocations"
          >
            <ul
              className="locaciones"
            />
          </div>
        </div>
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="input datapicker"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Check-in/ Check-out"
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <a
          href="/product"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
           
          <button
            className="button2"
          >
            Buscar
          </button>
        </a>
      </section>
    </div>
    <div
      className="searchCategory"
    >
      <section
        className="searchByType"
      >
        <h4>
          Buscar por tipo de alojamiento
        </h4>
        <p>
          Cargando Categorías...
        </p>
        <div
          className="categoryCards"
        />
      </section>
    </div>
    <div>
      <section
        className="recommendation"
      >
        <div
          className="recommendationContainer"
        >
          <h4
            className="recommendationtitle"
          >
            Recomendaciones
          </h4>
          <div
            className="recommendationCards"
          >
            <p>
              Cargando Recomendaciones...
            </p>
          </div>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <h6>
      ©2021 Digital Booking
    </h6>
    <div
      className="social"
    >
      <i
        className="fab fa-facebook"
      />
      <i
        className="fab fa-linkedin-in"
      />
      <i
        className="fab fa-twitter"
      />
      <i
        className="fab fa-instagram"
      />
    </div>
  </footer>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que aparezca la barra de búsqueda al hacer scroll 1`] = `null`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /administration renderiza el componente Forbidden. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="forbidden.png"
    />
    <h3
      className="title"
    >
      Login necesario
    </h3>
    <h5
      className="info"
    >
      Para acceder a esta página se requiere iniciar sesión.
    </h5>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="button"
      >
        Iniciar sesión
      </button>
    </a>
  </section>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /category/1 renderiza el componente Category. 1`] = `
<div
  className="cat"
>
  <section
    className="title"
  >
    <div
      className="titleContainer"
    >
      <h4
        className="category"
      />
      <i
        className="fas fa-arrow-left"
        onClick={[Function]}
      />
    </div>
  </section>
  <p>
    Cargando Recomendaciones...
  </p>
  <div
    className="recommendationContainer"
  >
    <section
      className="recommendation"
    >
      <h4
        className="recommendationtitle"
      >
        Recomendaciones de 
      </h4>
      <div
        className="recommendationCards"
      >
        <p>
          Cargando Locaciones...
        </p>
      </div>
    </section>
  </div>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /login renderiza el componente UserLogin. 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="login"
  >
    <h2>
      Iniciar sesión
    </h2>
    <div
      className="item"
    >
      <p>
        Correo electrónico
      </p>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="item"
    >
      <p>
        Contraseña
      </p>
      <input
        className="input"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <i
        className="fas fa-eye-slash"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      />
    </div>
    <h6
      className="errorm"
    />
    <button
      className="button"
      name="submit"
    >
      Ingresar
    </button>
    <div
      className="create"
    >
      <p>
        ¿Aun no tienes cuenta?
      </p>
      <a
        href="/register"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h5>
          Registrate
        </h5>
      </a>
    </div>
  </div>
</form>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /logout renderiza el componente Main. 1`] = `
<div
  className="App"
>
  Array [
    <div
      className="menuhide"
    >
      <div>
        <section
          className="header"
        >
          <button
            className="logout"
            onClick={[Function]}
          >
            X
          </button>
          <h4>
            MENÚ
          </h4>
        </section>
        <section
          className="buttons"
        >
          <a
            className="button"
            href="/register"
            onClick={[Function]}
          >
            <h5>
              Crear cuenta
            </h5>
          </a>
          <a
            className="button"
            href="/login"
            onClick={[Function]}
          >
            <h5>
              Iniciar sesión
            </h5>
          </a>
        </section>
      </div>
      <div
        className="social"
      >
        <section
          className="header"
        />
        <div
          className="socialbar"
        >
          <i
            className="fab fa-facebook"
          />
          <i
            className="fab fa-linkedin-in"
          />
          <i
            className="fab fa-twitter"
          />
          <i
            className="fab fa-instagram"
          />
        </div>
      </div>
    </div>,
    <nav
      className="color1"
    >
      <img
        alt="Logo"
        onClick={[Function]}
        src="logo.png"
      />
      <section
        className="searchInput"
      >
        <i
          className="fas fa-search"
        />
        <input
          className="searchNav"
          placeholder="Encontrá tu lugar"
          type="text"
        />
      </section>
      <div
        className="registerButtons"
      >
        <a
          href="/register"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
           
          <button
            className="button"
          >
            Crear cuenta
          </button>
        </a>
        <a
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
           
          <button
            className="button"
          >
            Iniciar sesion
          </button>
        </a>
      </div>
      <img
        alt="ícono menú"
        className="hamburguesita"
        onClick={[Function]}
        src="hamburguer.svg"
      />
    </nav>,
  ]
  <main
    className="main"
  >
    <div
      className="search"
    >
      <p>
        Busca ofertas en hoteles, casas y mucho más
      </p>
      <section>
        <div
          className="container"
        >
          <input
            className="input"
            content="width=device-width, initial-scale=1, maximum-scale=1"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="¿A dónde vamos?"
            type="text"
            value=""
          />
          <div
            className="containerLocations"
          >
            <ul
              className="locaciones"
            />
          </div>
        </div>
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="input datapicker"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Check-in/ Check-out"
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <a
          href="/product"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
           
          <button
            className="button2"
          >
            Buscar
          </button>
        </a>
      </section>
    </div>
    <div
      className="searchCategory"
    >
      <section
        className="searchByType"
      >
        <h4>
          Buscar por tipo de alojamiento
        </h4>
        <p>
          Cargando Categorías...
        </p>
        <div
          className="categoryCards"
        />
      </section>
    </div>
    <div>
      <section
        className="recommendation"
      >
        <div
          className="recommendationContainer"
        >
          <h4
            className="recommendationtitle"
          >
            Recomendaciones
          </h4>
          <div
            className="recommendationCards"
          >
            <p>
              Cargando Recomendaciones...
            </p>
          </div>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <h6>
      ©2021 Digital Booking
    </h6>
    <div
      className="social"
    >
      <i
        className="fab fa-facebook"
      />
      <i
        className="fab fa-linkedin-in"
      />
      <i
        className="fab fa-twitter"
      />
      <i
        className="fab fa-instagram"
      />
    </div>
  </footer>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /product/1 renderiza el componente SingleProduct. 1`] = `
<div>
  Loading...
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /register renderiza el componente Sign Up. 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="signup"
  >
    <h2>
      Crear cuenta
    </h2>
    <section
      className="fullname"
    >
      <div
        className="item"
      >
        <p>
          Nombre
        </p>
        <input
          className="input"
          name="nombre"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="item"
      >
        <p>
          Apellido
        </p>
        <input
          className="input"
          name="apellido"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </section>
    <div
      className="item"
    >
      <p>
        Correo electrónico
      </p>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="item"
    >
      <p>
        Contraseña
      </p>
      <input
        className="input"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <i
        className="fas fa-eye-slash"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      />
    </div>
    <div
      className="item"
    >
      <p>
        Confirmar contraseña
      </p>
      <input
        className="input"
        name="repassword"
        onChange={[Function]}
        type="password"
        value=""
      />
      <i
        className="fas fa-eye-slash"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      />
    </div>
    <h6
      className="errorm"
    />
    <button
      className="button"
      name="submit"
    >
      Registrate
    </button>
    <div
      className="create"
    >
      <p>
        ¿Ya tienes cuenta?
      </p>
      <a
        href="/login"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h5>
          Inicia sesión
        </h5>
      </a>
    </div>
  </div>
</form>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta /reservation/1 renderiza el componente Forbidden. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="forbidden.png"
    />
    <h3
      className="title"
    >
      Login necesario
    </h3>
    <h5
      className="info"
    >
      Para acceder a esta página se requiere iniciar sesión.
    </h5>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="button"
      >
        Iniciar sesión
      </button>
    </a>
  </section>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta cancellation renderiza el componente ResSuccess. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="checked.png"
    />
    <h3
      className="title"
    />
    <h5
      className="info"
    />
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
      />
    </a>
  </section>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta confirmation renderiza el componente ResSuccess. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="checked.png"
    />
    <h3
      className="title"
    />
    <h5
      className="info"
    />
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
      />
    </a>
  </section>
</div>
`;

exports[`Pruebas en componente <App /> Comprobar que la ruta verification renderiza el componente ResSuccess. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="checked.png"
    />
    <h3
      className="title"
    >
      ¡Muchas gracias!
    </h3>
    <h5
      className="info"
    >
      Su email ha sido verificado exitosamente.
    </h5>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
      >
        Volver al home
      </button>
    </a>
  </section>
</div>
`;
